demo/full/bxf-parser2/src/parser/schedule/schedule_elements/primary/index.js

```diff
@@ -0,0 +1,49 @@
+import reduceChildren from "../../../reduce_children";
+import { parsedTimeCode } from "../../../../utils/timecode";
+import parseTransition from "../../../node_commons/transition";
+import {
+  parseStartDateTime,
+  parseSmpteTimeCode,
+} from "../../../node_commons/time";
+
+export function parsePrimaryEvent(node) { // assuming primary is video stream
+
+  function parseLengthOption(node) {
+
+    function parseSmpteDuration(node) {
+      return reduceChildren(node, (res, name, node) => {
+        switch (name.toLowerCase()) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: "switch" statements should have at least 3 "case" clauses ([javascript:S1301](https://sonarcloud.io/organizations/rx-player/rules#rule_key=javascript%3AS1301))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=366&issues=AWbt7Z35W3Y69bz5aGXI&open=AWbt7Z35W3Y69bz5aGXI)

[ISSUE_KEY]: AWbt7Z35W3Y69bz5aGXI